jbmc/src/java_bytecode/README.md

```diff
@@ -256,9 +256,72 @@ To be documented.
 
 To be documented.
 
-\section java-bytecode-lazy-methods-v1 Context Insensitive lazy methods (aka lazy methods v1)
-
-To be documented.
+\section loading-strategies Loading strategies
+
+Loading strategies are policies that determine what classes and method bodies
+ are loaded. On an inital pass, the symbols for all classes in all paths on the 
+ Java classpath are loaded. Eager loading is a policy that then loads method 
+ bodies for every one of these classes and for all the method bodies in these
+ classes. Lazy loading strategies are policies that only load class symbols 
+ and/or method bodies when they are in some way requested.
+
+The mechanism used to achieve this is initially common to both eager and 
+ context-insensitive lazy loading. \ref java_bytecode_languaget::typecheck calls
+[java_bytecode_convert_class](\ref java_bytecode_languaget::java_bytecode_convert_class)
+(for each class loaded by the class loader) to create symbols for all classes
+and the methods in them. The methods, along with their parsed representation
+(including bytecode) are also added to a map called
+\ref java_bytecode_languaget::method_bytecode via a reference held in
+\ref java_bytecode_convert_classt::method_bytecode. typecheck then performs a
+switch over the value of
+[lazy_methods_mode](\ref java_bytecode_languaget::lazy_methods_mode) to 
+determine which loading strategy to use.
+
+\subsection eager-loading Eager loading
+
+If [lazy_methods_mode](\ref java_bytecode_languaget::lazy_methods_mode) is
+\ref java_bytecode_languaget::LAZY_METHODS_MODE_EAGER then eager loading is 
+used. Under eager loading
+\ref java_bytecode_languaget::convert_single_method(const irep_idt &, symbol_table_baset &)
+is called once for each method listed in method_bytecode (described above). This
+then calls
+\ref java_bytecode_languaget::convert_single_method(const irep_idt &, symbol_table_baset &, optionalt<ci_lazy_methods_neededt>)
+without a ci_lazy_methods_neededt object, which calls
+\ref java_bytecode_convert_method, passing in the method parse tree. This in
+turn uses \ref java_bytecode_convert_methodt to turn the bytecode into symbols
+for the parameters and local variables and finish populating the symbol for the
+method, including converting the instructions to a codet.
+
+\subsection java-bytecode-lazy-methods-v1 Context-Insensitive lazy methods (aka lazy methods v1)
+
+Context-insensitive lazy loading is an alternative method body loading strategy
+to eager loading that has been used in Deeptest for a while.
+
+Context-insensitive lazy loading starts at the method given by the `--function`
+ argument and follows type references (e.g. in the definitions of fields and 
+ method parameters) and function references (at function call sites) to 
+ locate further classes and methods to load.
+
+If [lazy_methods_mode](\ref java_bytecode_languaget::lazy_methods_mode) is
```

Comment:
This sentence seems out of place here. Would it be better at the beginning of this subsection? The first few sentences of this subsection don't really flow, like they were written separately and put together later.